lms/djangoapps/verify_student/views.py

```diff
@@ -1047,3 +1055,187 @@ def reverification_window_expired(_request):
+    """
+    @method_decorator(login_required)
+    def get(self, request, course_id, checkpoint_name):
+        """
+        display this view
+        """
+        user = request.user
+        course_key = CourseKey.from_string(course_id)
+        course = modulestore().get_course(course_key)
+        if course is None:
+            raise Http404
+        checkpoint = VerificationCheckpoint.get_verification_checkpoint(course_key, checkpoint_name)
+        if checkpoint is None:
+            raise Http404
+        init_verification = SoftwareSecurePhotoVerification.get_initial_verification(user)
+        if not init_verification:
+            return redirect(reverse('verify_student_verify_later', kwargs={'course_id': unicode(course_key)}))
+
+        display_steps = self._display_steps()
+        requirements = self._requirements(display_steps, request.user.is_active)
+        current_step = display_steps[0]['name']
+        expired_verified_course_mode, unexpired_paid_course_mode = PayAndVerifyView()._get_expired_verified_and_paid_mode(course_key)
+
+        context = {
+            'contribution_amount': 10000,
+            'course': course,
+            'course_key': unicode(course_key),
+            'course_mode': unexpired_paid_course_mode,
+            'courseware_url': "http:coursewareurl.com",
+            'current_step': current_step,
+            'disable_courseware_js': True,
+            'display_steps': display_steps,
+            'is_active': json.dumps(request.user.is_active),
+            'message_key': PayAndVerifyView.VERIFY_LATER_MSG,
+            'platform_name': settings.PLATFORM_NAME,
+            'purchase_endpoint': get_purchase_endpoint(),
+            'requirements': requirements,
+            'user_full_name': request.user.profile.name,
+            'verification_deadline': "",
+            'submit_photos_url': reverse('verify_student_incourse_reverification_confirmation')
+
+        }
+        return render_to_response("verify_student/incourse_reverify.html", context)
+
+    @method_decorator(login_required)
+    def post(self, request, course_id, checkpoint_name):
+        """
+        submits the reverification to SoftwareSecure
+        """
+        try:
+            user = request.user
+            course_id = CourseKey.from_string(course_id)
+            checkpoint = VerificationCheckpoint.get_verification_checkpoint(course_id, checkpoint_name)
+            if checkpoint is None:
+                raise VerificationCheckpointException
+            init_verification = SoftwareSecurePhotoVerification.get_initial_verification(user)
+            if not init_verification:
+                raise SoftwareSecurePhotoVerification.DoesNotExist
+            b64_face_image = request.POST['face_image'].split(",")[1]
+            attempt = SoftwareSecurePhotoVerification(user=request.user)
+            attempt.upload_face_image(b64_face_image.decode('base64'))
+            attempt.fetch_photo_id_image()
```

Comment:
I don't think this is going to fetch the right image, since it's using the mid-course reverification logic (calling `original_verification` instead of `get_initial_verification`.